tests/src/testcommon.hh

```diff
@@ -236,29 +232,42 @@ template <typename NonLinearOperator>
   return t;
 }
 
-template <Ikarus::ResultType resType>
-[[nodiscard]] auto checkCalculateAt(auto& nonLinearOperator, auto& fe, const auto& feRequirements,
+template <template <typename, int, int> class resType, bool voigt = true>
+[[nodiscard]] auto checkCalculateAt(auto& /*nonLinearOperator*/, auto& fe, const auto& feRequirements,
                                     const auto& expectedResult, const auto& evaluationPositions,
                                     const std::string& messageIfFailed = "") {
   Dune::TestSuite t("Test of the calulateAt function for " + Dune::className(fe));
 
+  using FiniteElement = std::remove_cvref_t<decltype(fe)>;
   Eigen::MatrixXd computedResults(expectedResult.rows(), expectedResult.cols());
-  for (int i = 0; const auto& pos : evaluationPositions) {
-    auto result              = fe.template calculateAt<resType>(feRequirements, pos);
-    computedResults.row(i++) = result.transpose();
-  }
 
-  const bool isResultCorrect = isApproxSame(computedResults, expectedResult, 1e-8);
-  t.check(isResultCorrect) << "Computed Result for " << toString(resType) << " is not the same as expected result:\n"
-                           << "It is:\n"
-                           << computedResults << "\nBut should be:\n"
-                           << expectedResult << "\n"
-                           << messageIfFailed;
+  if constexpr (FiniteElement::template canProvideResultType<resType>()) {
+    for (int i = 0; const auto& pos : evaluationPositions) {
+      if constexpr (voigt) {
+        auto result              = fe.template calculateAt<resType>(feRequirements, pos).asVec();
+        computedResults.row(i++) = result.transpose();
+      } else {
+        auto result = fe.template calculateAt<resType>(feRequirements, pos).asMat();
+        Eigen::Map<Eigen::VectorXd> vector(result.data(), result.size());
+        computedResults.row(i++) = vector.transpose();
+      }
+    }
+    const bool isResultCorrect = isApproxSame(computedResults, expectedResult, 1e-8);
+    t.check(isResultCorrect) << "Computed Result for " << Ikarus::toString<resType>()
+                             << " is not the same as expected result:\n"
+                             << "It is:\n"
+                             << computedResults << "\nBut should be:\n"
+                             << expectedResult << "\n"
+                             << messageIfFailed;
+  } else
+    t.check(false) << "Element can not provide the requested RsultType " << Ikarus::toString<resType>()
```

Comment:
Failing at runtime maybe does not make sense here. I imagine someone writes a new element and want to perform checks on it. Then this function is used in the tests. Then the developer could be warned that this check can not be performed for the new element at compile time, do tom some reason.